### PR TITLE
Non-functional fixes

### PR DESCRIPTION
**Pull request by cshung**
This PR is intended for making cosmetic fixes without making and functional changes.

- This eliminated a couple of accidental use of 'tab'.
- It fixes the wrong indentation in various places.
- Removed unwanted instrumentation code blocks.
```
COUNT_CYCLES
ENABLE_INSTRUMENTATION 
STAGE_STATS <- there’s actually no code for this
DEBUG_WRITE_WATCH
STRESS_PINNING
TIME_GC
TIME_WRITE_WATCH
```

### CODE REVIEW

**Comment by cshung** [comment]
@Maoni0 is added to the review. #Closed

File: src/coreclr/src/gc/gcpriv.h [hunk 1]
```diff
@@ -132,31 +132,22 @@ inline void FATAL_GC_ERROR()
...
-//#define TIME_WRITE_WATCH  //time GetWriteWatch and ResetWriteWatch calls
-//#define COUNT_CYCLES  //Use cycle counter for timing
 //#define JOIN_STATS         //amount of time spent in the join
 //also, see TIME_SUSPEND in switches.h.
 
 //#define SYNCHRONIZATION_STATS
 //#define SEG_REUSE_STATS
 
-#if defined (SYNCHRONIZATION_STATS) || defined (STAGE_STATS)
+#if defined (SYNCHRONIZATION_STATS)
```
Comment (Maoni0) [inline src/coreclr/src/gc/gcpriv.h]: nit - could be changed to 
`#ifdef SYNCHRONIZATION_STATS`
just to be consistent with the rest of the code

**Review by Maoni0 — APPROVED** [review]
other than that nit, the rest LGTM!

**Review by Maoni0 — APPROVED** [review]
LGTM, please be sure to squash the commits when merging. thanks!

**Pull request merged**